apis/v1/options.go

```diff
@@ -63,7 +63,9 @@ func (o *Options) UnmarshalJSON(b []byte) error {
 	if err := d.Decode(&entries); err != nil {
 		return err
 	}
-	o.parse(entries)
+	if err := o.parse(entries); err != nil {
+		return err
```

Comment:
is the return error propagated to the user at object creation?

Reply:
Not this particular one. But an error that says your body of your request is invalid.
Its the `"code": 400` in the webhook response.

Reply:
If the error is wrapped here would that be propagated to user? 

We should tell the user what was wrong in the CR if that is possible.

Reply:
oh seems i mixed things up. When you create Jeager with invalid `Options` it tells you whats wrong. 

```bash
$ k apply -f invalid.yaml                                                     
Error from server: error when creating "invalid.yaml": admission webhook "mjaeger.kb.io" denied the request: v1.Jaeger.Spec: v1.JaegerSpec.Strategy: Storage: v1.JaegerStorageSpec.Type: Options: unmarshalerDecoder: invalid option type, expect: string, got: map[string]interface {}, error found in #10 byte of ...|":"test"}},"type":"c|..., bigger context ...|"datacenter":"dev","enabled":false,"mode":"test"}},"type":"cassandra"},"strategy":"production"}}|...
```

Looking at the logs, we could explain error `400` a bit better. But placing a description in the  `"reason": ""` field needs to be done in `controller-runtime.webhook.webhooks`.